mlflow/utils/databricks_utils.py

```diff
@@ -439,34 +475,65 @@ def get_databricks_host_creds(server_uri=None):
         MlflowHostCreds which includes the hostname and authentication information necessary to
         talk to the Databricks server.
     """
+    # Since we do not record OAuth expiration time in OAuth file, perform periodic refresh
+    # of OAuth cache here. As currently configured (02/24) OAuth token in model serving environment
+    # guaranteed to have at least 30 min remaining on TTL at any point in time but refresh at higher
+    # rate here to be safe and in case those values change in the future.
+    OAUTH_CACHE_REFRESH_DURATION = 10 * 60
+    OAUTH_CACHE_ENV_VAR = "DATABRICKS_DEPENDENCY_OAUTH_CACHE"
+    OAUTH_CACHE_EXPIRATION_ENV_VAR = "DATABRICKS_DEPENDENCY_OAUTH_CACHE_EXIRY_TS"
+
     profile, path = get_db_info_from_uri(server_uri)
     config = ProfileConfigProvider(profile).get_config() if profile else get_config()
-    # if a path is specified, that implies a Databricks tracking URI of the form:
-    # databricks://profile-name/path-specifier
-    if (not config or not config.host) and path:
-        dbutils = _get_dbutils()
-        if dbutils:
-            # Prefix differentiates users and is provided as path information in the URI
-            key_prefix = path
-            host = dbutils.secrets.get(scope=profile, key=key_prefix + "-host")
-            token = dbutils.secrets.get(scope=profile, key=key_prefix + "-token")
-            if host and token:
-                config = DatabricksConfig.from_token(host=host, token=token, insecure=False)
-    if not config or not config.host:
-        _fail_malformed_databricks_auth(profile)
-
     insecure = hasattr(config, "insecure") and config.insecure
 
-    if config.username is not None and config.password is not None:
-        return MlflowHostCreds(
-            config.host,
-            username=config.username,
-            password=config.password,
-            ignore_tls_verification=insecure,
-        )
-    elif config.token:
-        return MlflowHostCreds(config.host, token=config.token, ignore_tls_verification=insecure)
-    _fail_malformed_databricks_auth(profile)
+    # if in model serving environment databricks attempt to fetch model dependency OAuth token
```

Comment:
Add a comment saying we refresh the cache once every 10 mins. Also can we make the period every 5 mins just to be overcautious?

Reply:
what are we concerned about? even 10 min should be far less than the 30 min ttl, and is likely to have greater reuse of cache

Reply:
Is there a good reason to diverge from what we did for lookup client? 
https://src.dev.databricks.com/databricks/universe/-/blob/feature-store/lookup-client/python/databricks/feature_store/lookup_engine/lookup_brickstore_http_engine.py?L28 

Keeping in sync will lead to much less configurations to keep track of

Reply:
ok we can do 5 min to keep consistent. Personally, I feel like we should do 30 min for both since its better cache performance and more intuitive to understand the TTL logic (Originally, I had the same questions that @WeichenXu123 asked.) But I think performance impact here should be negligible anyway.